refactor(toggle): follow current style guides and fix typescript issues

components is not currently following our current style guides and tsc is returning following errors

```
27       'bl-toggle': GenericTypes.IntrinsicCustomElement<ToggleProps> & GenericTypes.IntrinsicBoreElement<Props, Events>
                                                          ~~~~~~~~~~~

Toggle.tsx(27,56): error TS4033: Property ''bl-toggle'' of exported interface has or is using private name 'ToggleProps'.


27       'bl-toggle': GenericTypes.IntrinsicCustomElement<ToggleProps> & GenericTypes.IntrinsicBoreElement<Props, Events>
                                                                                                           ~~~~~

Toggle.tsx(27,105): error TS4033: Property ''bl-toggle'' of exported interface has or is using private name 'Props'.


27       'bl-toggle': GenericTypes.IntrinsicCustomElement<ToggleProps> & GenericTypes.IntrinsicBoreElement<Props, Events>
                                                                                                                  ~~~~~~

Toggle.tsx(27,112): error TS4033: Property ''bl-toggle'' of exported interface has or is using private name 'Events'.


32 export class Toggle extends Component<ToggleProps> {
                                         ~~~~~~~~~~~

Toggle.tsx(32,39): error TS4020: Extends clause of exported class 'Toggle' has or is using private name 'ToggleProps'.


34   static get props() {
                ~~~~~

Toggle.tsx(34,14): error TS4038: Return type of public static property getter from exported class has or is using name 'PropOptions' from external module "/Users/mario/ngParty/blaze-elements/node_modules/skatejs/src/ts-typings/api" but cannot be named.
```

--- a/packages/toggle/Toggle.tsx
+++ b/packages/toggle/Toggle.tsx
@@ -1,69 +1,79 @@
 import styles from './Toggle.scss';
-import { h, Component, prop, props, emit } from 'skatejs';
+import { h, Component, props, emit } from 'skatejs';
 import {
   ColorType,
   cssClassForColorType,
   css,
-  GenericTypes,
-  GenericEvents
+  GenericEvents,
+  prop,
+  shadyCssStyles
 } from '@blaze-elements/common';
 
-type ToggleProps = Props & EventProps;
-type EventProps = {
-  onChange?: GenericEvents.CustomChangeHandler<string>,
-};
-type Events = {
-  change?: GenericEvents.CustomChangeHandler<string>,
-};
-type Props = {
+export type ToggleProps = Props & EventProps;
+
+export type Attrs = {
   disabled?: boolean,
   checked?: boolean,
   color?: ColorType,
 };
 
-declare global {
-  namespace JSX {
-    interface IntrinsicElements {
-      'bl-toggle': GenericTypes.IntrinsicCustomElement<ToggleProps> & GenericTypes.IntrinsicBoreElement<Props, Events>
-    }
-  }
-}
+export type EventProps = {
+  onChange?: GenericEvents.CustomChangeHandler<string>,
+};
 
-export class Toggle extends Component<ToggleProps> {
-  static get is() { return 'bl-toggle'; }
-  static get props() {
-    return {
-      disabled: prop.boolean( {
-        attribute: true
-      } ),
-      checked: prop.boolean( {
-        attribute: true
-      } ),
-      color: prop.string( {
-        attribute: true
-      } )
-    };
-  }
+export type Events = {
+  change?: GenericEvents.CustomChangeHandler<string>,
+};
+
+export type Props = {
+  disabled?: boolean,
+  checked?: boolean,
+  color?: ColorType,
+};
+
+@shadyCssStyles()
+export default class Toggle extends Component<ToggleProps> {
+
   static get events() {
     return {
       CHANGE: 'change'
     };
   }
-  disabled = false;
-  checked = false;
-  color: ColorType;
+
+  @prop( {
+    type: Boolean,
+    attribute: {
+      source: true
+    }
+  } ) disabled = false;
+
+  @prop( {
+    type: Boolean,
+    attribute: {
+      source: true
+    }
+  } ) checked = false;
+
+  @prop( {
+    type: String,
+    attribute: {
+      source: true
+    }
+  } ) color: ColorType;
+
+  get css() { return styles; }
 
   connectedCallback() {
     super.connectedCallback();
     this.handleChecked = this.handleChecked.bind( this );
   }
+
   renderCallback() {
     const { disabled, checked, color } = this;
     const colorClass = cssClassForColorType( 'c-toggle', color );
     const className = css( 'c-toggle', colorClass );
 
-    return [
-      <style>{styles}</style>,
+    return (
       <label class={className}>
         <input
           type="checkbox"
@@ -76,7 +86,7 @@ export class Toggle extends Component<ToggleProps> {
         </div>
         <slot />
       </label>
-    ];
+    );
   }
   private handleChecked( event: Event ) {
     event.stopPropagation();

--- a/packages/toggle/index.demo.tsx
+++ b/packages/toggle/index.demo.tsx
@@ -1,9 +1,9 @@
 import { h, Component } from 'skatejs';
-import { Toggle } from './Toggle';
-import { GenericEvents } from '@blaze-elements/common';
+import { Toggle } from './index';
+import { GenericEvents, customElement } from '@blaze-elements/common';
 
-class Demo extends Component<void> {
-  static get is() { return 'bl-toggle-demo'; };
+@customElement( 'bl-toggle-demo' )
+export class Demo extends Component<void> {
   toggleChecked = false;
 
   constructor() {
@@ -17,8 +17,7 @@ class Demo extends Component<void> {
   }
 
   renderCallback() {
-    return [
-      <style />,
+    return (
       <fieldset>
         <legend>{Toggle.is}</legend>
 
@@ -29,9 +28,6 @@ class Demo extends Component<void> {
         </div>
 
       </fieldset>
-    ];
+    );
   }
 }
-
-
-customElements.define( Demo.is, Demo );

--- a/packages/toggle/index.ts
+++ b/packages/toggle/index.ts
@@ -1,5 +1,19 @@
-import { define } from 'skatejs';
-import { Toggle } from './Toggle';
 
-export { Toggle } from './Toggle';
-define( Toggle );
+import { customElement, GenericTypes } from '@blaze-elements/common';
+import RawToggle, { ToggleProps, Attrs, Events } from './Toggle';
+
+const Toggle = customElement( 'bl-toggle' )( RawToggle ) as typeof RawToggle;
+
+export {
+  Toggle
+};
+
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'bl-toggle': GenericTypes.IntrinsicCustomElement<ToggleProps>
+      & GenericTypes.IntrinsicBoreElement<Attrs, Events>
+    }
+  }
+}

